src/app/Fake.Tools.Octo/Octo.fs

```diff
@@ -214,96 +214,119 @@ let private deployCommandLine (opts:DeployReleaseOptions) =
-    |> List.fold (+) ""
+    |> List.filter String.isNotNullOrEmpty
 
 /// Maps a command to string input for the octopus tools cli.
 let private commandLine command =
     match command with
     | CreateRelease (opts, None) ->
-        sprintf " create-release%s" (releaseCommandLine opts)
+        "create-release" :: (releaseCommandLine opts)
     | CreateRelease (opts, Some (dopts)) ->
-        sprintf " create-release%s%s" (releaseCommandLine opts) (deployCommandLine dopts)
+        "create-release" :: ( List.append (releaseCommandLine opts) (deployCommandLine dopts) )
     | DeployRelease opts ->
-        sprintf " deploy-release%s" (deployCommandLine opts)
+        "deploy-release" :: (deployCommandLine opts)
     | DeleteReleases opts ->
-        sprintf " delete-releases%s" (deleteCommandLine opts)
+        "delete-releases" :: (deleteCommandLine opts)
     | ListEnvironments -> 
-        " list-environments" 
+        ["list-environments"] 
     | Push opts -> 
-        sprintf " push%s" (pushCommandLine opts)
+        "push" :: (pushCommandLine opts)
 
 let private exec command options =
-    
     let serverCommandLineForTracing (opts: ServerOptions) = 
         serverCommandLine { opts with ApiKey = "(Removed for security purposes)" }
 
     let tool = options.ToolPath @@ options.ToolName
-    let args = commandLine command |>(+)<| serverCommandLine options.Server
-    let traceArgs = commandLine command |>(+)<| serverCommandLineForTracing options.Server
+    let args = List.append (commandLine command) (serverCommandLine options.Server)
+               |> Arguments.OfArgs
+    let traceArgs = (List.append (commandLine command) (serverCommandLineForTracing options.Server)) |> List.fold (+) ""
     
     let commandString = command.ToString()
 
-    use __ = Trace.traceTask "Octo "commandString
+    use __ = Trace.traceTask "Octo " commandString
     Trace.trace (tool + traceArgs)
         
     let result = 
-        Process.execSimple (fun info -> 
-            {info with 
-                Arguments = args
-                WorkingDirectory = options.WorkingDirectory
-                FileName = tool
-            }
-        ) options.Timeout
+        RawCommand (tool, args)
+        |> CreateProcess.fromCommand
+        |> CreateProcess.withWorkingDirectory options.WorkingDirectory
+        |> CreateProcess.withTimeout options.Timeout
+        |> Proc.run
+        |> (fun finishedProccess -> finishedProccess.ExitCode)
 
     match result with
-    | 0 -> ()
-    | _ -> failwithf "Octo %s failed. Process finished with exit code %i" commandString result
-
-/// Creates a release.
-let createRelease setParams = 
+    | 0 ->
+        __.MarkSuccess()
+        result
+    | _ ->
+        __.MarkFailed()
+        failwithf "Octo %s failed. Process finished with exit code %i" commandString result
+        result
+
+/// Creates a release and returns the exit code.
+let createReleaseWithExitCode setParams = 
```

Comment:
Are you sure this is doing what you intended? From a quick look it looks like this either returns `0` or throws an exception (without actually testing it)

Reply:
I hadn't tested that yet but looking at it again after the weekend, yes this seems wrong.